ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/helpers/BlurTransformation.kt

```diff
@@ -0,0 +1,68 @@
+import android.content.Context
+import android.graphics.Bitmap
+import android.renderscript.Allocation
+import android.renderscript.Element
+import android.renderscript.RenderScript
+import android.renderscript.ScriptIntrinsicBlur
+import coil.size.Size
+import coil.transform.Transformation
+import kotlin.math.min
+
+/**
+ * BlurTransformation class applies a blur on a given Bitmap image.
+ * The blurring is performed with RenderScript, and should only be used in API level < 31, since newer versions
+ * of Android have their own native blur implementation.
+ *
+ * @property radius - The radius of the square used for blurring. Higher values
+ *                    produce a more pronounced blur effect.
+ */
+internal class BlurTransformation(
+    private val context: Context,
+    private val radius: Float,
+) : Transformation {
+
+    override val cacheKey: String = "${javaClass.name}-$radius"
+
+    override suspend fun transform(
+        input: Bitmap,
+        size: Size,
+    ): Bitmap {
+        return input.blur(context, radius)
+    }
+
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        return other is BlurTransformation && radius == other.radius
+    }
+
+    override fun hashCode(): Int = radius.hashCode()
+}
+
+// max radius supported by RenderScript
+private const val MAX_SUPPORTED_RADIUS = 25
+
+internal fun Bitmap.blur(context: Context, radius: Float = 25f): Bitmap {
```

Comment:
Maybe? Or we could also define `MAX_SUPPORTED_RADIUS` as a float.
```suggestion
internal fun Bitmap.blur(context: Context, radius: Float = MAX_SUPPORTED_RADIUS.toFloat()): Bitmap {
```

Reply:
Oh thanks I forgot to use it here too.